src/test/java/integ/global_attribute/GlobalAttributeProcess.java

```diff
@@ -0,0 +1,214 @@
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableSet;
+import io.xdb.core.command.CommandRequest;
+import io.xdb.core.command.CommandResults;
+import io.xdb.core.communication.Communication;
+import io.xdb.core.context.Context;
+import io.xdb.core.exception.GlobalAttributeNotFoundException;
+import io.xdb.core.persistence.Persistence;
+import io.xdb.core.persistence.PersistenceSchema;
+import io.xdb.core.persistence.PersistenceTableRowToUpsert;
+import io.xdb.core.persistence.PersistenceTableSchema;
+import io.xdb.core.process.Process;
+import io.xdb.core.process.ProcessOptions;
+import io.xdb.core.process.ProcessStartConfig;
+import io.xdb.core.state.AsyncState;
+import io.xdb.core.state.AsyncStateOptions;
+import io.xdb.core.state.StateDecision;
+import io.xdb.core.state.StateSchema;
+import io.xdb.gen.models.AttributeWriteConflictMode;
+import org.springframework.stereotype.Component;
+
+@Component
+public class GlobalAttributeProcess implements Process {
+
+    public static final String TABLE_NAME = "sample_user_table";
+    public static final String PK_KEY = "user_id";
+    public static final String PK_VALUE = "pk_value";
+    public static final String COL_KEY_1 = "first_name";
+    public static final String COL_VALUE_1 = "col_1_value";
+    public static final String COL_VALUE_1_2 = "col_1_value_2";
+    public static final String COL_KEY_2 = "last_name";
+    public static final String COL_VALUE_2 = "col_2_value";
+    public static final String COL_KEY_3 = "create_timestamp";
+    public static final String COL_VALUE_3 = "111";
+
+    @Override
+    public StateSchema getStateSchema() {
+        return StateSchema.withStartingState(
+            new GlobalAttributeProcessStartingState(),
+            new GlobalAttributeProcessNextState1(),
+            new GlobalAttributeProcessNextState2()
+        );
+    }
+
+    @Override
+    public PersistenceSchema getPersistenceSchemaToLoad() {
+        return PersistenceSchema.withGlobalAttributes(
+            PersistenceTableSchema.withPrimaryKey(TABLE_NAME, PK_KEY, ImmutableSet.of(COL_KEY_1, COL_KEY_2))
+        );
+    }
+
+    @Override
+    public ProcessOptions getOptions() {
+        return ProcessOptions
+            .builder(GlobalAttributeProcess.class)
+            .processStartConfig(
+                ProcessStartConfig
+                    .builder()
+                    .globalAttributesToUpsert(
+                        ImmutableList.of(
+                            PersistenceTableRowToUpsert
+                                .createWithPrimaryKeyColumn(
+                                    TABLE_NAME,
+                                    PK_KEY,
+                                    PK_VALUE,
+                                    AttributeWriteConflictMode.RETURNERRORONCONFLICT
```

Comment:
ahhh no... this enum looks so bad here.. 😮‍💨

Reply:
I think I know why -- 
Let's fix the xdb-api to use under scores instead(like ProcessIdReusePolicy) 😨

Reply:
Done updated in https://github.com/xdblab/xdb/pull/86
https://github.com/xdblab/xdb-apis/pull/19
and https://github.com/xdblab/xdb-golang-sdk/pull/37